### PR TITLE
[MODEL-1474] Quality modes as list

### PR DESCRIPTION
**Pull request by lb-kamilk**
# Description

Allow creation of projects as both benchmark and consensus. Can only be rolled out after B&C GA.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document change (fix typo or modifying any markdown files, code comments or anything in the examples folder only)

## All Submissions

- [x] Have you followed the guidelines in our Contributing document?
- [x] Have you provided a description?
- [x] Are your changes properly formatted?

## New Feature Submissions

- [ ] Does your submission pass tests?
- [ ] Have you added thorough tests for your new feature?
- [ ] Have you commented your code, particularly in hard-to-understand areas?
- [ ] Have you added a Docstring?

## Changes to Core Features

- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully run tests with your changes locally?
- [x] Have you updated any code comments, as applicable?

### CODE REVIEW

**Review by vbrodsky — CHANGES_REQUESTED** [review]
I think making queue_mode from QualityModel to List[...] makes it incompatible for sdk users. You need to be able to take in either one or the whole list

File: examples/project_configuration/project_setup.ipynb [hunk 1]
```diff
@@ -118,7 +118,7 @@
     },
     {
       "metadata": {},
-      "source": "batch_project = client.create_project(\n    name=\"Project Setup Demo\",\n    quality_mode=QualityMode.\n    Consensus,  # For benchmarks use quality_mode = QualityMode.Benchmark\n    media_type=lb.MediaType.Image,\n)\n\nbatch_project.setup_editor(ontology)",
+      "source": "batch_project = client.create_project(\n    name=\"Project Setup Demo\",\n    quality_modes=[QualityMode.Consensus\n                  ],  # For benchmarks use quality_mode = QualityMode.Benchmark\n    media_type=lb.MediaType.Image,\n)\n\nbatch_project.setup_editor(ontology)",
```
Comment (vbrodsky) [inline examples/project_configuration/project_setup.ipynb:121]: nice you did it

File: libs/labelbox/src/labelbox/client.py [hunk 2]
```diff
@@ -751,6 +751,8 @@ def create_project(self, **kwargs) -> Project:
             queue_mode (Optional[QueueMode]): The queue mode to use
             quality_mode (Optional[QualityMode]): The quality mode to use (e.g. Benchmark, Consensus). Defaults to
```
Comment (vbrodsky) [inline libs/labelbox/src/labelbox/client.py:769]: can we document that this attr will be deprecated in a future version. And also print a deprecation version if this attr is used

File: libs/labelbox/src/labelbox/client.py [hunk 3]
```diff
@@ -891,22 +893,30 @@ def _create_project(self, **kwargs) -> Project:
...
+                "Cannot use both quality_modes and quality_mode at the same time. Use one or the other.")
+
+        if not quality_modes and not quality_mode:
+            logger.info("Defaulting quality modes to Benchmark.")
 
         data = kwargs
+        data.pop("quality_modes", None)
         data.pop("quality_mode", None)
-        if quality_mode is None or quality_mode is QualityMode.Benchmark:
+        if quality_modes is None or quality_modes == [QualityMode.Benchmark] or quality_mode is QualityMode.Benchmark:
```
Comment (vbrodsky) [inline libs/labelbox/src/labelbox/client.py]:
```suggestion
        if quality_modes is None or len(quality_modes) == 0 or quality_modes == [QualityMode.Benchmark] or quality_mode is QualityMode.Benchmark:
```

**Review by vbrodsky — DISMISSED** [review]

**Review by vbrodsky — DISMISSED** [review]

File: libs/labelbox/src/labelbox/client.py [hunk 4]
```diff
@@ -751,6 +751,8 @@ def create_project(self, **kwargs) -> Project:
             queue_mode (Optional[QueueMode]): The queue mode to use
             quality_mode (Optional[QualityMode]): The quality mode to use (e.g. Benchmark, Consensus). Defaults to
                 Benchmark
+            quality_modes (Optional[List[QualityMode]]): The quality modes to use (e.g. Benchmark, Consensus). Defaults to
```
Comment (vbrodsky) [inline libs/labelbox/src/labelbox/client.py:771]: looking at the code below, wondering whether this should be a Set, not a List

File: libs/labelbox/src/labelbox/client.py [hunk 5]
```diff
@@ -891,22 +893,45 @@ def _create_project(self, **kwargs) -> Project:
...
+            data["auto_audit_percentage"] = CONSENSUS_AUTO_AUDIT_PERCENTAGE
+            data["is_benchmark_enabled"] = True
+            data["is_consensus_enabled"] = True
+        elif quality_modes == [QualityMode.Benchmark] or quality_mode is QualityMode.Benchmark:
             data[
                 "auto_audit_number_of_labels"] = BENCHMARK_AUTO_AUDIT_NUMBER_OF_LABELS
             data["auto_audit_percentage"] = BENCHMARK_AUTO_AUDIT_PERCENTAGE
-        elif quality_mode is QualityMode.Consensus:
+            data["is_benchmark_enabled"] = True
+        elif quality_modes == [QualityMode.Consensus] or quality_mode is QualityMode.Consensus:
```
Comment (vbrodsky) [inline libs/labelbox/src/labelbox/client.py]: perhaps we could wrap `quality_mode` in a List to make this code cleaner

**Review by vbrodsky — COMMENTED** [review]
Left two comments for what I see as improvements, let me know what you think
Also are you planning to update sdk integration tests now or in the future

File: libs/labelbox/src/labelbox/client.py (hunk 4 above)
Comment (lb-kamilk) [reply libs/labelbox/src/labelbox/client.py:771]: Changed the code to accept either a list or a set, casting to set eventually

**Review by vbrodsky — DISMISSED** [review]

**Review by vbrodsky — APPROVED** [review]

**Pull request merged**